v0.7/speech_recognition/rnnt/accuracy_eval.py

```diff
@@ -31,13 +31,19 @@ def main():
     hypotheses = []
     references = []
     for result in results:
-        hypotheses.append(array.array('q', bytes.fromhex(result["data"])).tolist())
+        hypotheses.append(array.array('b', bytes.fromhex(result["data"])).tolist())
         references.append(manifest[result["qsl_idx"]]["transcript"])
-    hypotheses = __gather_predictions([hypotheses], labels=labels)
+
+    # Convert ASCII output into string
+    for idx in range(len(hypotheses)):
+        hypotheses[idx] = ''.join([chr(c) for c in hypotheses[idx]])
```

Comment:
What was `__gather_predictions()` doing? Is it doing this? (not sure if this will work for int64 output)

Reply:
Originally, It's assuming that the output is arranged in [' ', 'a', 'b',..., 'z'] and translate output index into string. But the actual output is changed to ASCII instead of index.

Reply:
Then I think we shouldn't change the default behavior here. The output should still be indices (but can be in INT8, INT16, INT32, or INT64).

Reply:
I'm going to need to think more about this one tomorrow. 

I'm worried about making a change that might be incorrect with our rules. Allowing different integer dtypes for the output seems harmless, though, I would also like to double-check on that.

Reply:
Our current rules don't whitelist any post-processing steps (even though this one seems harmless), which is why I'm a bit iffy about things: https://github.com/mlperf/inference_policies/blob/master/inference_rules.adoc#71-pre--and-post-processing

Reply:
@nvzhihanj I think we should use `hypotheses = __gather_predictions([hypotheses], labels=labels)`, and just cast the elements inside `hypotheses` with `int()` before passing it into `__gather_predictions`.

@galv I will bring this up in the WG. In my opinion, we shouldn't force everyone to use INT64 for indices.

@DilipSequeira FYI.

Reply:
Addressed

Reply:
@galv We changed it such that we don't touch how the accuracy is computed, but only allow more output dtypes for the token indices. Would you be okay with that? Thanks

Reply:
It certainly sounds reasonable. However, I suggest bringing it up at the Tuesday morning meeting. I'm happy to make an issue in inference_rules repo about this. FYI @christ1ne 

Presumably other classification tasks like imagenet have had the same ambiguity.

Reply:
I wrote an issue here: https://github.com/mlperf/inference_policies/issues/158

Hopefully this can get resolved tomorrow morning, after which we can merge this.